psi4/src/psi4/libfock/jk.h

```diff
@@ -1185,127 +1185,28 @@ class PSI_API MemDFJK : public JK {
-    /// Coulomb Metric
-    SharedMatrix J_metric_;
-    /// per-thread TwoBodyAOInt object (for computing three-center ERIs)
-    std::vector<std::shared_ptr<TwoBodyAOInt>> eri_computers_;
-
-    // => Semi-Numerical Stuff <= //
-
-    /// Small DFTGrid for initial SCF iterations
-    std::shared_ptr<DFTGrid> grid_init_;
-    /// Large DFTGrid for the final SCF iteration
-    std::shared_ptr<DFTGrid> grid_final_;
-    /// Overlap fitting metric for grid_initial_
-    SharedMatrix Q_init_;
-    /// Overlap fitting metric for grid_final_
-    SharedMatrix Q_final_;
-
-    std::string name() override { return "DFJCOSK"; }
-    size_t memory_estimate() override;
-
-    // => Required Algorithm-Specific Methods <= //
-
-    /// Do we need to backtransform to C1 under the hood?
-    bool C1() const override { return true; }
-    /// Setup integrals, files, etc
-    void preiterations() override;
-    /// Compute J/K for current C/D
-    void compute_JK() override;
-    /// Delete integrals, files, etc
-    void postiterations() override;
-    /// Set up Incfock variables per iteration
-    void incfock_setup();
-    /// Post-iteration Incfock processing
-    void incfock_postiter();
-
-    /// Build the coulomb (J) matrix
-    void build_J(std::vector<std::shared_ptr<Matrix> >& D,
-                 std::vector<std::shared_ptr<Matrix> >& J);
-
-    /// Build the exchange (K) matrix
-    void build_K(std::vector<std::shared_ptr<Matrix> >& D,
-                 std::vector<std::shared_ptr<Matrix> >& K);
-
-    /// Common initialization
-    void common_init();
-
-   public:
-    // => Constructors < = //
-
-    /**
-     * @param primary primary basis set for this system.
-     *        AO2USO transforms will be built with the molecule
-     *        contained in this basis object, so the incoming
-     *        C matrices must have the same spatial symmetry
-     *        structure as this molecule
-     */
-    DFJCOSK(std::shared_ptr<BasisSet> primary, std::shared_ptr<BasisSet> auxiliary, Options& options);
-    /// Destructor
-    ~DFJCOSK() override;
-
-    // => Knobs <= //
-    bool do_incfock_iter() { return do_incfock_iter_; }
-
-    /**
-    * Print header information regarding JK
-    * type on output file
-    */
-    void print_header() const override;
-
-    /**
-     * Clear D_prev_
-     */
-    void clear_D_prev() { D_prev_.clear();}
-};
-
-/**
- * Class DFJLinK
+ * JK implementation framework enabling arbitrary mixing and matching
+ * of separate J and K construction algorithms.
+ * Current algorithms in place:
+ * J: Direct DF-J
+ * K: COSX, LinK
  *
- * JK implementation using the direct density-fitted coulomb algorithm of Weigend
- * and the Linear Exchange algorithm of Ochsenfeld
- * TODO: Combine DFJLink with DFJCOSX to start CompositeJK
+ * TODO: Implement SplitJK companion framework for truly arbitrary mixing and matching
  */
-class PSI_API DFJLinK : public JK {
+class PSI_API CompositeJK : public JK {
```

Comment:
The diff interpreted the CompositeJK implementation as a modification of DFJLinK.

The new CompositeJK class is essentially a union of the DFJLinK and DFJCOSK classes, with small changes to support the new front end.